pkgs/jovian-greeter/greeter.py

```diff
@@ -0,0 +1,211 @@
+# A minimal greetd greeter that runs a user's preferred session
+# in $HOME/.local/state/steamos-session-select
+
+import json
+import logging
+import os
+import re
+import socket
+import struct
+import subprocess
+import sys
+from pathlib import Path
+from typing import Any, Optional, Iterable, Mapping, List
+
+DEFAULT_SESSION = 'steam-wayland'
+HELPER_PREFIX = Path('/run/current-system/sw/lib/jovian-greeter')
+
+class Session:
+    TYPE = 'tty'
+
+    def __init__(self, name: str, path: Path):
+        self.name = name
+        with open(path, 'r') as f:
+            self.content = f.read()
+
+    def get_command(self) -> Optional[List[str]]:
+        if command := self._get_property('Exec'):
+            return command.split(' ')
+
+        return None
+
+    def get_environment(self) -> List[str]:
+        envs = [
+            f'XDG_SESSION_TYPE={self.TYPE}',
+            f'XDG_SESSION_DESKTOP={self.name}',
+        ]
+
+        if desktop_names := self._get_property('DesktopNames'):
+            envs.append(f'XDG_CURRENT_DESKTOP={desktop_names}')
+
+        return envs
+
+    def _get_property(self, property: str) -> Optional[str]:
+        if matches := re.search(f'^{property}=(.*)$', self.content, re.MULTILINE):
+            return matches.group(1)
+
+        return None
+
+class WaylandSession(Session):
+    TYPE = 'wayland'
+
+class XSession(Session):
+    TYPE = 'X11'
```

Comment:
Trying to switch to an X11 session was failing with this error:
`greetd[2228714]: pam_systemd(greetd:session): Failed to create session: Invalid session type X11`
Switching to a lower case X resolved the issue
```suggestion
    TYPE = 'x11'
```